src/backend/x11/mod.rs

```diff
@@ -162,8 +163,30 @@ impl X11Backend {
         let (connection, screen_number) = XConnection::new(&logger)?;
         let connection = Arc::new(connection);
         info!(logger, "Connected to screen {}", screen_number);
-
         let xcb = connection.xcb_connection();
+
+        // Does the X server support dri3?
+        let (dri3_major, dri3_minor) = {
+            let extension = xcb.query_extension(dri3::X11_EXTENSION_NAME.as_bytes())?.reply()?;
+
+            if !extension.present {
+                todo!("Missing extension")
+            }
+
+            // TODO: Figure out what on earth xcb does with these 2 params?
+            let version = xcb.dri3_query_version(1, 2)?.reply()?;
```

Comment:
Since you asked on irc... well:

Here are some docs: https://github.com/freedesktop/xorg-dri3proto/blob/master/dri3proto.txt#L64

And here is the implementation: https://gitlab.freedesktop.org/xorg/xserver/-/blob/18d3131f9a332096825c09106a931c427246ddb7/dri3/dri3_request.c#L45

I don't really know xorg code, but it seems like this checks which protocol version the server (I guess this depends on the graphics driver?) can support. It then sends you this version. It does not do anything else.

You are checking for `major_version < 1`. As far as I can tell, there was never a 0.x version of dri3, so this should never trigger.